### DISCUSSION TITLE
Save relational username to collection

### DISCUSSION THREAD

**Discussion by MuslemRahimi** [pocketbase/pocketbase#2226] [Q&A] [answered]
Hi,

This question is kinda hard for me to explain.

I have a collection called "posts" where one of the fields is "user" that is relational. I want to give in my sveltekit app the "user" as locals.user.username so that pocketbase saves the username to it. At the moment it saves the userID. How can i make it work to the username instead of userID? 

![Screenshot from 2023-04-05 23-05-45](https://user-images.githubusercontent.com/22890183/230211751-f4888030-ff6b-4c0b-9626-65e75653b2bb.png)

**Comment by ganigeorgiev** [comment, marked as answer]
If `posts.user` is a regular `text` field, then you should be able to store whatever string value you want.

If `posts.user` is a `relation` field then you can't set a non-id value.

Why do you want to store the username in the first place? If you want to access the associated user to the `posts` record, you can check [Expanding relations](https://pocketbase.io/docs/expanding-relations/), for example:
```js
const post = await pb.collection("posts").getOne("POST_ID", { expand: "user" })
```

**Reply by MuslemRahimi** [reply]
thanks!
